test/mbedclient/utest/m2mresource/test_m2mresource.cpp

```diff
@@ -91,7 +91,13 @@ void Test_M2MResource::test_base_type()
 
 void Test_M2MResource::test_muliptle_instances()
 {
+    m2mbase_stub::sn_resource = (M2MBase::lwm2m_parameters_s*) malloc(sizeof(M2MBase::lwm2m_parameters_s));
+    memset(m2mbase_stub::sn_resource,0,sizeof(M2MBase::lwm2m_parameters_s));
```

Comment:
Is there a reason for getting these temp lwm2m_parameter structs from heap? Why not use a local variable and set the m2mbase_stub::sn_resource to point to that? Then the libc would also find out, if the code tried to mistakenly free the given struct.
Another way to reduce code would be to use calloc().